xmr-btc/src/bob.rs

```diff
@@ -279,8 +341,9 @@ pub struct State2b {
     tx_refund_encsig: EncryptedSignature,
 }
 
-impl State2b {
-    pub async fn watch_for_lock_xmr<W>(self, xmr_wallet: &W, msg: alice::Message2) -> Result<State3>
+impl State3 {
+    // todo: loop until punish? timelock has expired
```

Comment:
:wrench: It would be the refund/cancel timelock in this case. But in any case, I would remove this TODO as we don't yet know in which layer we want to do this.

Reply:
correct!